src/gui/screen_printing_serial.cpp

```diff
@@ -75,50 +104,46 @@ void screen_printing_serial_init(screen_t *screen) {
         WINDOW_CLS_ICON, root,
         rect_ui16((240 - pt_ico.x) / 2, gui_defaults.msg_box_sz.y, pt_ico.x, pt_ico.y),
         &(pw->octo_icon));
-    //window_set_color_back(id, COLOR_GRAY);
     window_enable(id);
     window_set_icon_id(id, IDR_PNG_serial_printing);
     pw->octo_icon.win.f_enabled = 0;
     pw->octo_icon.win.f_disabled = 0;
 
-    //button tune
-    const rect_ui16_t rect_bt_tune = rect_ui16(8, 185, 64, 64);
-    id = window_create_ptr(
-        WINDOW_CLS_ICON, root,
-        rect_bt_tune,
-        &(pw->tune_button));
-    window_set_color_back(id, COLOR_GRAY);
-    window_enable(id);
-    window_set_icon_id(id, IDR_PNG_menu_icon_settings);
+    for (unsigned int col = 0; col < 3; col++) {
+        id = window_create_ptr(
+            WINDOW_CLS_ICON, root,
+            rect_ui16(8 + (15 + 64) * col, 185, 64, 64),
+            &(pw->w_buttons[col]));
+        window_set_color_back(id, COLOR_GRAY);
+        window_set_tag(id, col + 1);
+        window_enable(id);
+
+        id = window_create_ptr(
+            WINDOW_CLS_TEXT, root,
+            rect_ui16(80 * col, 196 + 48 + 8, 80, 22),
+            &(pw->w_labels[col]));
+        pw->w_labels[col].font = resource_font(IDR_FNT_SMALL);
+        window_set_padding(id, padding_ui8(0, 0, 0, 0));
+        window_set_alignment(id, ALIGN_CENTER);
+    }
 
-    //button tune text
-    id = window_create_ptr(
-        WINDOW_CLS_TEXT, root,
-        rect_ui16(0, 196 + 48 + 8, 80, 22),
-        &(pw->tune_label));
-    pw->tune_label.font = resource_font(IDR_FNT_SMALL);
-    window_set_padding(id, padding_ui8(0, 0, 0, 0));
-    window_set_alignment(id, ALIGN_CENTER);
-    window_set_text(id, "Tune");
-    window_set_tag(id, Tag_bt_tune);
-    window_set_focus(id);
-
-    //text
-    const point_ui16_t pt_txt = font_meas_text(resource_font(IDR_FNT_NORMAL), txt0);
-    const rect_ui16_t rect_txt = rect_ui16(
-        (240 + rect_bt_tune.x + rect_bt_tune.w - pt_txt.x) / 2,
-        gui_defaults.msg_box_sz.y + pt_ico.y,
-        240 - (240 + rect_bt_tune.x + rect_bt_tune.w - pt_txt.x) / 2, //cannot use 240-rect_txt.x .. stupid C
-        95);
-    id = window_create_ptr(WINDOW_CLS_TEXT, root, rect_txt, &(pw->w_message));
-    window_set_alignment(id, ALIGN_LEFT_TOP);
-    //window_set_padding(id, padding_ui8(0, 2, 0, 2));
-    window_set_text(id, txts[0]);
+    // -- CONTROLS
+    window_icon_t *sp_button;
+    // -- tune button
+    sp_button = &pw->w_buttons[BUTTON_TUNE];
+    set_icon_and_label(iid_tune, sp_button->win.id, pw->w_labels[BUTTON_TUNE].win.id);
+    // -- pause
+    sp_button = &pw->w_buttons[BUTTON_PAUSE];
+    set_icon_and_label(iid_pause, sp_button->win.id, pw->w_labels[BUTTON_PAUSE].win.id);
+    // -- disconnect
+    sp_button = &pw->w_buttons[BUTTON_DISCONNECT];
```

Comment:
static_assert( BUTTON_DISCONNECT < number_of_buttons );
because this will fail at compile time warning the user of the array being too short